dev/Licensing/MsixLicensing.cpp

```diff
@@ -6,11 +6,92 @@
 
 #include <WindowsAppRuntime-License.h>
 
-STDAPI MsixInstallLicenses() noexcept try
+STDAPI MsixInstallLicenses(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag) noexcept try
 {
-    //TODO
-    // FOREACH file IN pkgdir\MSIX\*_license.xml
-    //  InstallLicenseFile(file)
-    return S_OK;
+    // Licenses are only needed for Stable releases (versionTag=null or "")
+    if (versionTag && (versionTag[0] != L'\0'))
+    {
+        return S_OK;
+    }
+
+    const UINT16 majorVersion{ HIWORD(majorMinorVersion) };
+    const UINT16 minorVersion{ LOWORD(majorMinorVersion) };
+    WCHAR packageFamilyName[PACKAGE_FAMILY_NAME_MAX_LENGTH + 1]{};
+    wsprintf(packageFamilyName, L"Microsoft.WindowsAppRuntime.%hu.%hu_8wekyb3d8bbwe", majorVersion, minorVersion);
+
+    // Find a framework package in the family (any will do)
+    winrt::Windows::Management::Deployment::PackageManager packageManager;
+    winrt::hstring currentUser;
+    const auto c_packageTypes{ winrt::Windows::Management::Deployment::PackageTypes::Framework };
+    for (auto package : packageManager.FindPackagesForUserWithPackageTypes(currentUser, packageFamilyName, c_packageTypes))
+    {
+        // Install the license(s)
+        const auto packageId{ package.Id() };
+        const auto packageFullName{ packageId.FullName() };
+        RETURN_IF_FAILED(MsixInstallLicensesInPackage(packageFullName.c_str()));
+        return S_OK;
+    }
+
+    // No framework package found in the family
+    RETURN_HR_MSG(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), "PackageFamilyName:%ls", packageFamilyName);
 }
 CATCH_RETURN();
+
+STDAPI MsixInstallLicensesInPackage(PCWSTR frameworkPackageFullName) noexcept try
+{
+    // Get the package's path
+    //
+    // NOTE: Use GetStagedPackagePathByFullName as don't care if the package is registered.
+    //
+    // NOTE: Don't use GetStagedPackagePathByFullName2 as we want the *install* location
+    //       (not e.g. modifiable location). Plus Get...2() is new in 20H1 (too new for us)
+    //       where Get...() exists down to Win81. In this case simplest is bestest.
+    wil::unique_cotaskmem_string packagePathBufferDynamic;
+    uint32_t packagePathLength{};
+    const auto rc{ GetStagedPackagePathByFullName(frameworkPackageFullName, &packagePathLength, nullptr) };
+    if (rc != ERROR_INSUFFICIENT_BUFFER)
+    {
+        RETURN_HR_MSG(HRESULT_FROM_WIN32(rc), "PackageFullName:%ls", frameworkPackageFullName);
+    }
+    auto packagePath{ wil::make_cotaskmem_string(nullptr, packagePathLength) };
+    RETURN_IF_WIN32_ERROR(GetStagedPackagePathByFullName(frameworkPackageFullName, &packagePathLength, packagePath.get()));
```

Comment:
Consider lifting this block into a helper function, then eventually into WIL, like `wil::cotaskmem_string GetStagedPackagedPath(PCWSTR packageFullName)`

Reply:
... which you already have below, as `GetPathForPackage` ? Reuse it here, put a definition into a header used by this directory.

Reply:
There's a bunch of refactoring and optimizing that'll happen soon, just not wise to do today. We'll pull this work into main/1.1 (and work out some merge issues) and _then_ refactor a bit, when time isn't so crunched.